src/sentry/auth/authenticators/u2f.py

```diff
@@ -43,6 +47,23 @@ def _get_url_prefix() -> str:
     return options.get("system.url-prefix")
 
 
+def _valid_staff_timestamp(request, limit: timedelta = STAFF_AUTH_FLOW_MAX_AGE) -> bool:
+    """
+    Returns whether or not the staff timestamp exists and is valid within the
+    timedelta. If the timestamp is invalid, it is removed from the session.
+    """
+    timestamp = request.session.get("staff_auth_flow")
+    if not timestamp:
+        return False
+
+    flag_datetime = datetime.utcfromtimestamp(timestamp)
+    if datetime.now() - flag_datetime > limit:
+        del request.session["staff_auth_flow"]
```

Comment:
I'm always wary of deleting keys, especially on request metadata. Is this absolutely necessary?

Reply:
Yes this is fine.